src/V3Param.cpp

```diff
@@ -552,17 +552,17 @@ class ParamProcessor final {
         // Recursion may need level cleanups
         if (newmodp->level() <= m_modp->level()) newmodp->level(m_modp->level() + 1);
         if ((newmodp->level() - srcModp->level()) >= (v3Global.opt.moduleRecursionDepth() - 2)) {
-            cellp->v3error("Exceeded maximum --module-recursion-depth of "
+            cellp->v3fatal("Exceeded maximum --module-recursion-depth of "
```

Comment:
Can we keep this an error?  I appreciate it might cause other downstream errors, but user-caused problems generally should not be fatal as gives a bad experience.

Reply:
As per https://github.com/verilator/verilator/pull/3541/commits/0a22f28838ae629796db28e1f04a6a57cfb40d03 commit message, I discovered that when this error is reached, Verilator gets stuck in some infinite loop and its memory footprint steadily increases (presumably, I kept it running on the simplest test case for an hour when I killed it because it took 2 GiB of RAM already).

This happened with debug mode turned off, turning it on makes it output the same error over and over again until reaching the limit (sorry for not mentioning that more directly in the commit message), which is arguably better, but if it were not for the error limit, I guess it would also loop, spamming the same error forever.

Another solution I tried was to add a `return` statement, but this results in a UASSERT failing in the calling function, which is also fatal.

Reply:
I'd prefer the return and later fatal, so at least it indicates this specific message correctly.